dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -42,26 +42,51 @@ def find_toxicity(role, text, locale, level_id)
       details = nil
       start_time = Time.now
       report_openai_safety_check("Start")
-      # Try twice in case of network errors or model not correctly following directions and
-      # replying with something other valid expected output.
-      attempts = 1
-      Retryable.retryable(tries: 2) do
-        messages = safety_check_messages(text, level_id)
-        response = client.request_chat_completion(messages, 1)
-        raise "OpenAI request failed with status #{response.code}: #{response.body}" unless response.success?
-
-        evaluation = JSON.parse(response.body)['choices'][0]['message']['content']
-        unless VALID_EVALUATION_RESPONSES_SIMPLE.include?(evaluation)
+      attempts = 0
+      messages = safety_check_messages(text, level_id)
+
+      # Retry only on network-related exceptions
+      response = Retryable.retryable(
+        tries: 2,
+        on: [Net::OpenTimeout, Net::ReadTimeout, SocketError, Errno::ECONNRESET]
+      ) do
+        attempts += 1
+        client.request_chat_completion(messages, 1)
+      end
+      raise "OpenAI request failed with status #{response.code}: #{response.body}" unless response.success?
+
+      evaluation = JSON.parse(response.body)['choices'][0]['message']['content']
+      unless VALID_EVALUATION_RESPONSES_SIMPLE.include?(evaluation)
+        report_openai_safety_check("InvalidResponse")
+        attempts +=1
+
+        # Fallback to structured call (non-retryable)
+        response = request_structured_safety_check(text, level_id)
+        raise "OpenAI structured request failed with status #{response.code}: #{response.body}" unless response.success?
+
+        body = JSON.parse(response.body)
+        raw_content = body.dig("choices", 0, "message", "content")
+
+        begin
+          parsed = JSON.parse(raw_content)
+        rescue JSON::ParserError
```

Comment:
Why don't we just let Ruby throw this error (vs rescuing and raising a custom error)?